Mathlib/CategoryTheory/FiberedCategory/Fibered.lean

```diff
@@ -72,4 +93,95 @@ lemma pullbackObj_proj : p.obj (pullbackObj ha f) = R :=
 
 end IsPreFibered
 
+namespace IsFibered
+
+open IsCartesian IsPreFibered
+
+/-- In a fibered category, any cartesian morphism is strongly cartesian. -/
+instance isStronglyCartesian_of_isCartesian (p : 𝒳 ⥤ 𝒮) [p.IsFibered] {R S : 𝒮} (f : R ⟶ S)
+    {a b : 𝒳} (φ : a ⟶ b) [p.IsCartesian f φ] : p.IsStronglyCartesian f φ where
+  universal_property' := by
+    intro a' g φ' hφ'
+    -- Let `ψ` be a cartesian arrow lying over `g`
+    let ψ := pullbackMap (domain_eq p f φ) g
+    -- Let `τ` be the map induced by the universal property of `ψ ≫ φ`.
+    let τ := IsCartesian.map p (g ≫ f) (ψ ≫ φ) φ'
+    use τ ≫ ψ
+    -- It is easily verified that `τ ≫ ψ` lifts `g` and `τ ≫ ψ ≫ φ = φ'`
+    refine ⟨⟨inferInstance, by simp only [assoc, IsCartesian.fac, τ]⟩, ?_⟩
+    -- It remains to check that `τ ≫ ψ` is unique.
+    -- So fix another lift `π` of `g` satisfying `π ≫ φ = φ'`.
+    intro π ⟨hπ, hπ_comp⟩
+    -- Write `π` as `π = τ' ≫ ψ` for some `τ'` induced by the universal property of `ψ`.
+    rw [← fac p g ψ π]
+    -- It remains to show that `τ' = τ`. This follows again from the universal property of `ψ`.
+    congr 1
+    apply map_uniq
+    rwa [← assoc, IsCartesian.fac]
+
+/-- In a category which admits strongly cartesian pullbacks, any cartesian morphism is
+strongly cartesian. This is a helper-lemma for the fact that admitting strongly cartesian pullbacks
+implies being fibered. -/
+lemma isStronglyCartesian_of_exists_isCartesian' (p : 𝒳 ⥤ 𝒮) (h : ∀ (a : 𝒳) (R : 𝒮)
+    (f : R ⟶ p.obj a), ∃ (b : 𝒳) (φ : b ⟶ a), IsStronglyCartesian p f φ) {R S : 𝒮} (f : R ⟶ S)
+      {a b : 𝒳} (φ : a ⟶ b) [p.IsCartesian f φ] : p.IsStronglyCartesian f φ := by
+  constructor
+  intro c g φ' hφ'
+  subst_hom_lift p f φ; clear a b R S
+  -- Let `ψ` be a cartesian arrow lying over `g`
+  obtain ⟨a', ψ, hψ⟩ := h _ _ (p.map φ)
+  -- Let `τ' : c ⟶ a'` be the map induced by the universal property of `ψ`
+  let τ' := IsStronglyCartesian.map p (p.map φ) ψ (f':= g ≫ p.map φ) rfl φ'
+  -- Let `Φ : a' ≅ a` be natural isomorphism induced between `φ` and `ψ`.
+  let Φ := domainUniqueUpToIso p (p.map φ) φ ψ
+  -- The map induced by `φ` will be `τ' ≫ Φ.hom`
+  use τ' ≫ Φ.hom
+  -- It is easily verified that `τ' ≫ Φ.hom` lifts `g` and `τ' ≫ Φ.hom ≫ φ = φ'`
+  refine ⟨⟨by simp only [Φ]; infer_instance, ?_⟩, ?_⟩
+  · simp [τ', Φ, IsStronglyCartesian.map_uniq p (p.map φ) ψ rfl φ']
+  -- It remains to check that it is unique. This follows from the universal property of `ψ`.
+  intro π ⟨hπ, hπ_comp⟩
+  rw [← Iso.comp_inv_eq]
+  apply IsStronglyCartesian.map_uniq p (p.map φ) ψ rfl φ'
+  simp [hπ_comp, Φ]
+
+
+/-- Alternate constructor for `IsFibered`, a functor `p : 𝒳 ⥤ 𝒴` is fibered if any diagram of the
+form
+```
+          a
+          -
+          |
+          v
+R --f--> p(a)
+```
+admits a strongly cartesian lift `b ⟶ a` of `f`. -/
+lemma of_exists_isStronglyCartesian' {p : 𝒳 ⥤ 𝒮} (h : ∀ (a : 𝒳) (R : 𝒮) (f : R ⟶ p.obj a),
```

Comment:
This lemma should be in the `Functor.IsFibered` namespace. Also, I am not sure we need `'` in the name.

Reply:
I changed the whole `IsFibered` namespace to `Functor.IsFibered` in the end